pkg/loop/internal/relayer/pluginprovider/ext/median/test/datasource.go

```diff
@@ -46,13 +53,34 @@ func (s staticDataSource) Observe(ctx context.Context, timestamp types.ReportTim
 	return s.Value, nil
 }
 
+type CompareError struct {
+	Got      *big.Int
+	Expected *big.Int
+}
+
+func (e *CompareError) Error() string {
+	return fmt.Sprintf("expected Value %s but got %s", e.Expected, e.Got)
+}
+
+func (e *CompareError) GotZero() bool {
+	return e.Got.Uint64() == 0
```

Comment:
nit/ Could tighten this up a bit:
```suggestion
	return e.Got.IsUint64() && e.Got.Uint64() == 0
```